Core.ApplicationServices/Model/EventHandler/ContractDeletedAdvicesHandler.cs

```diff
@@ -0,0 +1,27 @@
+﻿using System.Linq;
+using Core.DomainModel.Advice;
+using Core.DomainModel.ItContract.DomainEvents;
+using Core.DomainServices;
+using Infrastructure.Services.DomainEvents;
+
+namespace Core.ApplicationServices.Model.EventHandler
+{
+    public class ContractDeletedAdvicesHandler : IDomainEventHandler<ContractDeleted>
+    {
+        private readonly IGenericRepository<Advice> _adviceRepository;
+        private readonly IAdviceService _adviceService;
+
+        public ContractDeletedAdvicesHandler(IGenericRepository<Advice> adviceRepository, IAdviceService adviceService)
+        {
+            _adviceRepository = adviceRepository;
+            _adviceService = adviceService;
+        }
+
+        public void Handle(ContractDeleted domainEvent)
+        {
+            var contractDeleted = domainEvent.DeletedContract;
+            var toBeDeleted = _adviceRepository.Get(a => a.RelationId == contractDeleted.Id && a.Type == ObjectType.itContract).ToList();
```

Comment:
Måske bedre at hive fat i IAdviceRepository interfaces og så udbyde metoder til at grave de forskellige advice typer frem ;-). Ellers skal den der query vedligeholdes i alle dine handlers og hører de virkelig hjemme der (application layer)? 

I samme ombæring skal du nok ind og fjerne metoden GetOrhans på samme repository, for den skal så ikke bruges længere